invenio_pidrelations/contrib/versioning.py

```diff
@@ -28,146 +28,142 @@
-        return self.get_children(ordered=False).filter(
-                PIDRelation.index.isnot(None),
-                PersistentIdentifier.status == PIDStatus.RESERVED).order_by(
-                    PIDRelation.index.desc()).one_or_none()
+        """Get the draft (RESERVED) child."""
+        return super(PIDNodeVersioning, self).children.status(
+            PIDStatus.RESERVED
+        ).one_or_none()
 
     @property
     def draft_child_deposit(self):
-        """
-        Get the deposit of the draft child.
+        """Get the deposit PID of the draft child.
 
-        Return `None` if no new-version deposit exists.
+        Return `None` if no draft child PID exists.
         """
-        from invenio_pidrelations.contrib.records import RecordDraft
         if self.draft_child:
-            return RecordDraft.get_draft(self.draft_child)
+            return PIDNodeDraft(self.draft_child).children.one_or_none()
         else:
             return None
 
     def insert_draft_child(self, child):
         """Insert a draft child to versioning."""
+        if child.status != PIDStatus.RESERVED:
+            raise PIDRelationConsistencyError(
+                "Draft child should have status 'RESERVED'")
+
         if not self.draft_child:
             with db.session.begin_nested():
-                super(PIDVersioning, self).insert_child(child, index=-1)
+                super(PIDNodeVersioning, self).insert_child(child, index=-1)
         else:
-            raise Exception(
+            raise PIDRelationConsistencyError(
                 "Draft child already exists for this relation: {0}".format(
                     self.draft_child))
 
     def remove_draft_child(self):
         """Remove the draft child from versioning."""
         if self.draft_child:
             with db.session.begin_nested():
-                super(PIDVersioning, self).remove_child(self.draft_child,
-                                                        reorder=True)
+                super(PIDNodeVersioning, self).remove_child(self.draft_child,
+                                                            reorder=True)
 
     def update_redirect(self):
-        """Update the parent redirect to the current last child."""
-        if self.last_child:
-            if self.parent.status == PIDStatus.RESERVED:
-                self.parent.register()
-            self.parent.redirect(self.last_child)
+        """Update the parent redirect to the current last child.
 
-    @property
-    def children(self):
-        """Children of the parent."""
-        return self.get_children(pid_status=PIDStatus.REGISTERED, ordered=True)
+        This method should be called on the parent PID node.
+
+        Use this method when the status of a PID changed (ex: draft changed
+        from RESERVED to REGISTERED)
+        """
+        if self.last_child:
+            if self._resolved_pid.status == PIDStatus.RESERVED:
+                self._resolved_pid.register()
```

Comment:
I don't think we need these two lines as we will set the current PID status to REDIRECTED.